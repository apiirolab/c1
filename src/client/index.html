<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>