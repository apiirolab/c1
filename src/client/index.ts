export const info = 'Client Info';
