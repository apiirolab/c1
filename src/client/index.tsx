import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
