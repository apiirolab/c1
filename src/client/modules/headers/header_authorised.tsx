import * as React from 'react';

import logo = require('./logo.png');

import { Input, InputProps, Menu, MenuProps } from 'semantic-ui-react';
import styled, { StyledComponentClass } from 'styled-components';

import { Link, withRouter } from 'react-router-dom';
import { LogoutMenu } from '../login/logout_menu';
import { NotificationAlertContainer } from '../notifications/notification_alert';
import { Logo } from './header_logo';

const TopPanel = styled.div`
  background: #efefef;
`;

const Search: StyledComponentClass<InputProps, {}> = styled(Input)`
  width: 300px !important;
  margin: 3px;
`;

const TopMenu: StyledComponentClass<MenuProps, {}> = styled(Menu)`
  &&&&& {
    border: 0px;
    border-radius: 0px;
    background: inherit;
    box-shadow: inherit;
    margin-bottom: 0px;
  }
  &&&&& .item {
    color: #777;
  }
  .wideMenu {
    @media (max-width: 950px) {
      display: none !important;
    }
  }
`;

const Narrow = styled.div`
  @media (min-width: 950px) {
    display: none;
  }
`;

const BottomMenu: StyledComponentClass<MenuProps, {}> = styled(Menu)`
  &&&&& {
    margin-top: 0px;
  }

  &&&&& .item {
    color: #777;
  }
`;

const SearchMenuItem = styled(Menu.Item)`
  padding-left: 0px !important;
  margin-left: 0px !important;
`;

const MenuElements = () => (
  <>
    <Menu.Item
      as={Link}
      to="/process/create"
      name="home"
      content="Create"
      icon="plus"
      color="blue"
      active={location.pathname === '/process/create'}
    />
    <Menu.Item
      as={Link}
      to="/process/running"
      name="running"
      content="Running"
      active={!!location.pathname.match(/\/running/)}
    />
    <Menu.Item
      as={Link}
      to="/process/finished"
      name="finished"
      content="Finished"
      active={!!location.pathname.match(/\/finished/)}
    />
    <Menu.Item
      as={Link}
      to="/process/all"
      name="all"
      content="All"
      active={!!location.pathname.match(/\/all/)}
    />

    {/* <Menu.Item name="statistics" content="Statistics" /> */}
  </>
);

export const HeaderAuthorised = withRouter(({ location }) => (
  <TopPanel>
    <TopMenu secondary>
      <Menu.Item>
        <Link to="/">
          <Logo src={logo} alt="logo" />
        </Link>
      </Menu.Item>
      <SearchMenuItem>
        <Search icon="search" placeholder="Find Process..." />
      </SearchMenuItem>
      <Menu.Menu className="wideMenu">
        <MenuElements />
      </Menu.Menu>
      <Menu.Menu position="right">
        <NotificationAlertContainer active={!!location.pathname.match(/\/notifications/)} />
        <LogoutMenu>
          <Menu.Item icon="user" content="Profile" />
        </LogoutMenu>
      </Menu.Menu>
    </TopMenu>
    <Narrow>
      <BottomMenu secondary borderless pointing color="blue">
        <MenuElements />
      </BottomMenu>
    </Narrow>
    {/* <BottomMenu secondary borderless pointing color="blue">
          <Menu.Item
            as={Link}
            to="/process/create"
            name="home"
            content="Create"
            color="blue"
            active={location.pathname === '/process/create'}
          />
          <Menu.Item
            name="running"
            content="Running"
            active={location.pathname.match(/\/running/)}
          />
          <Menu.Item
            name="finished"
            content="Finished"
            active={location.pathname.match(/\/finished/)}
          />
          <Menu.Item name="all" content="All" active={location.pathname.match(/\/all/)} />

          <Menu.Menu position="right">
            <NotificationAlertContainer active={!!location.pathname.match(/\/notifications/)} />
            <Menu.Item name="favourites" content="Favourites" icon="star" />
          </Menu.Menu>
        </BottomMenu> */}
  </TopPanel>
));
