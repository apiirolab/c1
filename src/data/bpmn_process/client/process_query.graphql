query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
