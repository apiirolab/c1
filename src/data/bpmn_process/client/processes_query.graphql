query Processes($input: BpmnProcessesInput!) {
  processes(input: $input) {
    id
    name
  }
}
