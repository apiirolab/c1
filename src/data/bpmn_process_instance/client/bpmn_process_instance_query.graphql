fragment BpmnProcessResource on Resource {
  name
  type
  id
  content
}

fragment BpmnProcessComment on Comment {
  id
  date
  replyTo
  text
  user {
    name
  }
}

fragment BpmnProcessTask on BpmnTaskInstance {
  name
  dateStarted
  dateFinished
  performer {
    name
  }
  performerRoles
}

fragment BpmnProcessDefinition on BpmnProcess {
  name
  description
  type
  model
  resources {
    ...BpmnProcessResource
  }
}

query BpmnProcessInstanceQuery($id: String!) {
  bpmnProcessInstance(id: $id) {
    id
    status
    process {
      ...BpmnProcessDefinition
    }
    owner {
      name
    }
    tasks {
      ...BpmnProcessTask
    }
    comments {
      ...BpmnProcessComment
    }
    resources
    dateStarted
    dateFinished
  }
}
