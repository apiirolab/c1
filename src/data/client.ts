import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
