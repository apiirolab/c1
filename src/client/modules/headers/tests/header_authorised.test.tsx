import * as React from 'react';

import { create } from 'react-test-renderer';
import { Menu } from 'semantic-ui-react';

jest.mock('../../notifications/notification_alert', () => ({
  NotificationAlertContainer: () => <Menu.Item>0</Menu.Item>
}));

import { MockedProvider } from 'client/tests';
import { HeaderAuthorised } from '../header_authorised';

describe('Headers', () => {
  describe('Authorised', () => {
    const Header = (
      <MockedProvider>
        <HeaderAuthorised />
      </MockedProvider>
    );

    it('renders correctly', () => {
      const root = create(Header);
      expect(root).toMatchSnapshot();
    });

    return {
      component: Header
    };
  });
});
