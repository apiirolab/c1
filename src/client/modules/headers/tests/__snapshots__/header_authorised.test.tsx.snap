// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headers Authorised renders correctly 1`] = `
<div
  className="sc-bwzfXH kRBEAi"
>
  <div
    className="ui secondary sc-bxivhb ftwuBk menu"
  >
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="sc-bdVaJa kPmfWy"
          src="file"
        />
      </a>
    </div>
    <div
      className="item sc-EHOje cmurTc"
      onClick={[Function]}
    >
      <div
        className="ui icon input sc-htpNat cIYGEn"
      >
        <input
          onChange={[Function]}
          placeholder="Find Process..."
          type="text"
        />
        <i
          aria-hidden="true"
          className="search icon"
        />
      </div>
    </div>
    <div
      className="right menu"
    />
  </div>
  <div
    className="ui blue borderless pointing secondary sc-ifAKCX cEuSRI menu"
  >
    <a
      className="blue item"
      href="/process/create"
      onClick={[Function]}
    >
      Create
    </a>
    <div
      className="item"
      onClick={[Function]}
    >
      Running
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      Finished
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      All
    </div>
    <div
      className="right menu"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        0
      </div>
      <div
        className="item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="star icon"
        />
        Favourites
      </div>
    </div>
  </div>
</div>
`;
